### PR TITLE
Cleanup and Refactor

### PR DESCRIPTION
**Pull request by christopherjbaker**
* Improved workflow
* Updated to Node 18
* Renamed Rules to Configs
* Standardized order of values in each Config
* Added `import` config
* Improved Test structure

### CODE REVIEW

File: angular.js [hunk 1]
```diff
@@ -1,22 +1,12 @@
...
+    "./configs/rxjs",
+    "./configs/ngrx",
+    "./configs/jest",
+    "./configs/karma",
+    "./configs/angular-a11y",
+    "./configs/angular-best-practices",
   ].map(require.resolve),
-  rules: {
-    "no-unused-vars": "off",
-    "@typescript-eslint/no-unused-vars": [
```
Comment (christopherjbaker) [inline angular.js:14]: These were moved to TypeScript since both Angular and React had it.

File: configs/angular-best-practices.js [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
       files: ["*.html"],
       plugins: ["@angular-eslint/template"],
       rules: {
-        "@angular-eslint/template/banana-in-box": "error",
         "@angular-eslint/template/button-has-type": "warn",
         "@angular-eslint/template/conditional-complexity": "warn",
         "@angular-eslint/template/cyclomatic-complexity": "warn",
+        "@angular-eslint/template/no-negated-async": "warn",
+        "@angular-eslint/template/use-track-by-function": "warn",
+
```
Comment (m-thompson-code) [inline configs/angular-best-practices.js:14]: extra newline

File: configs/angular-a11y.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
+
   overrides: [
     {
       files: ["*.html"],
       plugins: ["@angular-eslint/template"],
       rules: {
+        "@angular-eslint/template/click-events-have-key-events": "warn",
+        "@angular-eslint/template/mouse-events-have-key-events": "warn",
+        "@angular-eslint/template/no-autofocus": "warn",
+
```
Comment (m-thompson-code) [inline configs/angular-a11y.js:12]: Extra newline? Were these updated manually or through prettier or something?

**Review by m-thompson-code — APPROVED** [review]
Looks good to me, left some minor nitpicks on newlines. Out of curiosity were these files updated manually or through some formatter?

None of my comments are blockers, +1

File: configs/angular-best-practices.js (hunk 2 above)
Comment (christopherjbaker) [reply configs/angular-best-practices.js:14]: I split the Off, Warn, and Error rules, with a newline between them.

File: configs/angular-a11y.js (hunk 3 above)
Comment (christopherjbaker) [reply configs/angular-a11y.js:12]: Not extra, on purpose. See other comment. =]

**Comment by christopherjbaker** [comment]
> Out of curiosity were these files updated manually or through some formatter?
I did add Prettier to help cleanup, but the sorting, grouping, and order standardization was all manual.

**Pull request merged**